libkbfs/folder_block_ops.go

```diff
@@ -2489,6 +2499,16 @@ func (fbo *folderBlockOps) searchForNodesInDirLocked(ctx context.Context,
 		return 0, err
 	}
 
+	// getDirLocked may have unlocked blockLock, which means the cache
+	// could have changed out from under us.  Verify that didn't
+	// happen, so we can avoid messing it up with nodes from an old MD
+	// version.  If it did happen, return a special error that lets
+	// the caller know they should retry with a fresh cache.
```

Comment:
Do we need to support multiple simultaneous node caches? If not, we can store it inside folderBlockOps. Additionally, is the problem here that `currDir.path` wouldn't be updated but `cache` would? How can `getDirLocked` update the cache from under us if we didn't pass the `cache` to it?